monorepo-migration/sync-focus.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+set -ex
+
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
+CURRENT_REPO_PATH="$(dirname -- "$SCRIPT_DIR")"
+
+REPO_NAME_TO_SYNC='focus-ios'
+MAIN_BRANCH_NAME='main'
+
+CURRENT_BETA_VERSION=$(grep -A 1 'BITRISE_BETA_VERSION' bitrise.yml | awk '/-/{getline; print}' | tr -d "'")
+CURRENT_MAJOR_VERSION="$(( CURRENT_MAJOR_VERSION + 1 ))"
+CURRENT_RELEASE_VERSION="$(( CURRENT_BETA_VERSION - 1 ))"
+BRANCHES_TO_SYNC_ON_CURRENT_REPO=("$MAIN_BRANCH_NAME" "releases_v$CURRENT_BETA_VERSION" "releases_v$CURRENT_RELEASE_VERSION")
+BRANCHES_TO_SYNC_ON_TMP_REPO=("$MAIN_BRANCH_NAME" "releases_v$CURRENT_BETA_VERSION.0.0" "releases_v$CURRENT_RELEASE_VERSION.0.0")
+PREP_BRANCHES=("$REPO_NAME_TO_SYNC-prep" "$REPO_NAME_TO_SYNC-prep-$CURRENT_BETA_VERSION" "$REPO_NAME_TO_SYNC-prep-$CURRENT_RELEASE_VERSION")
+
+TMP_REPO_PATH="/tmp/git/$REPO_NAME_TO_SYNC"
+TMP_REPO_BRANCH_NAME='TODO-what-is-the-branch'
```

Comment:
Still figuring out that bit here, seems this is used to fetch the specified branch `TMP_REPO_BRANCH_NAME` from the `TMP_REPO_PATH`. I think we need to create this branch? As a way maybe to freeze the state of that repository before we migrate it? @JohanLorenzo

Reply:
Ah, I remember! `TMP_REPO_BRANCH_NAME` is the branch where you will push the `main` branch + any commits you need for the migration. This branch will live on https://github.com/mozilla-mobile/focus-ios. I recommend a name like `focus-ios-migration`